modules/onboarding/main.tf

```diff
@@ -0,0 +1,79 @@
+#-----------------------------------------------------------------------------------------
+# Fetch the data sources
+#-----------------------------------------------------------------------------------------
+
+# TODO: this needs to be updated once datasources are available
+locals {
+  sysdig_tenancy_ocid          = "ocid1.tenancy.oc1..aaaaaaaa26htcit3eytwicf3gavyqkgwr54cmdhuo3iim6i2vfetelnbayha"
+  sysdig_onboarding_group_ocid = "ocid1.group.oc1..aaaaaaaanutpcrbz5yklzmkqlnsfc36eyaav5i7rfuxmdiyglgtgyj3jelmq"
+  sysdig_onboarding_user_ocid  = "ocid1.user.oc1..aaaaaaaajdn4twgdk4cxcx3alzrn6apuims2dlr2iqaritg764xim6l5jqea"
+}
+
+// compartment data to populate policies if onboarding a compartment
+data "oci_identity_compartment" "compartment" {
+  count = var.compartment_ocid != "" ? 1 : 0
+  id    = var.compartment_ocid
+}
+
+
+// random suffix for policy name
+resource "random_id" "suffix" {
+  byte_length = 3
+}
+
+#-----------------------------------------------------------------------------------------
+# Admit policy to allow Sysdig Tenant to read resources
+#-----------------------------------------------------------------------------------------
+
+resource "oci_identity_policy" "admit_onboarding_policy" {
+  name           = "AdmitSysdigSecureTenantOnboarding-${random_id.suffix.hex}"
+  description    = "Onboarding policy to allow inspect compartments in tenant/compartment"
+  compartment_id = var.tenancy_ocid
+  statements = [
+    "Define tenancy sysdigTenancy as ${local.sysdig_tenancy_ocid}",
+    "Define group onboardingGroup as ${local.sysdig_onboarding_group_ocid}",
+      var.compartment_ocid != "" ?
+      "Admit group onboardingGroup of tenancy sysdigTenancy to inspect compartments in compartment ${data.oci_identity_compartment.compartment[0].name}"
```

Comment:
I think the intent to tightly scope to only `inspect` is good. But I think its ok to `read` compartments especially since we might want to read tags, associated policies, timestamps & other compartment related metadata in future, which we can't w/ `inspect`.

Reply:
Got it! I can change it to `read` then - I went for the `inspect` since we talked about using the less permisive policy on the onboarding policy

Reply:
done